app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt

```diff
@@ -453,7 +452,7 @@ class MeetingViewModel(
   private fun addVideoTrack(track: HMSVideoTrack, peer: HMSPeer) {
     synchronized(_tracks) {
       // Check if this track already exists
-      val _track = _tracks.find { it.video == null && it.peer.peerID == peer.peerID }
+      val _track = _tracks.find {track.source != HMSTrackSource.SCREEN && it.peer.peerID == peer.peerID }
```

Comment:
Why would screen shares be ignored for add?

Reply:
We are not ignoring `screenShare` .. we are creating a new `MeetingTrack` object, which means creating a new tile in the UI for screenshare. 
If its a non screen share track getting added then we would create a new tile only if a tile does not exist for that peer